src/runtime_src/core/tools/xbreplay/CMakeLists.txt

```diff
@@ -0,0 +1,75 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved
+# -----------------------------------------------------------------------------
+# Include the generated header files (e.g., version.h)
+include_directories(${XRT_BINARY_DIR}/gen)
+
+set(XBREPLAY_NAME "xbreplay")
+
+if(WIN32)
+  set(XRT_HELPER_SCRIPTS "xbreplay" "xbreplay.bat")
+else()
+  set(XRT_HELPER_SCRIPTS "xbreplay")
+endif()
+
+set(PATH_PREFIX src)
+set(REPLAY_UTILS_DIR ${PATH_PREFIX}/utils)
+set(REPLAY_ENG_DIR ${PATH_PREFIX}/replay_eng)
+set(REPLAY_XRT_DIR ${PATH_PREFIX}/replay_xrt)
+set(SEQ_RECONSTRUCTOR_DIR ${PATH_PREFIX}/seq_reconstructor)
+
+include_directories (
+    ${SEQ_RECONSTRUCTOR_DIR}
+    ${REPLAY_ENG_DIR}
+    ${REPLAY_XRT_DIR}
+    ${PATH_PREFIX}
+)
+
+set(SRCS src/main.cpp
+   src/utils/cmd_args.cpp
+   src/utils/message.cpp
+   src/seq_reconstructor/seq_reconstructor.cpp
+   src/replay_eng/replay.cpp
+   src/replay_xrt/replay_xrt_bo.cpp
+   src/replay_xrt/replay_xrt_device.cpp
+   src/replay_xrt/replay_xrt_hwctx.cpp
+   src/replay_xrt/replay_xrt_kernel.cpp
+   src/replay_xrt/replay_xrt_run.cpp
+   src/replay_xrt/replay_xrt_xclbin.cpp)
+
+add_executable(${XBREPLAY_NAME} ${SRCS})
+
+# Static build is a Linux / Ubuntu option only
+if (XRT_STATIC_BUILD)
+  add_executable(${XBREPLAY_NAME}_static ${SRCS})
+  target_link_options(${XBREPLAY_NAME}_static PRIVATE "-static" "-L${Boost_LIBRARY_DIRS}")
+  # Bypass FindBoost versions and just link explicitly with boost libraries
+  # The -static link option will pick the static libraries.
+  target_link_libraries(${XBREPLAY_NAME}_static
+    PRIVATE
+    xrt_coreutil_static
+    boost_system
+    boost_program_options
+    -Wl,--whole-archive rt pthread -Wl,--no-whole-archive
+    uuid
+    dl
+    )
+  set_target_properties(${XBREPLAY_NAME}_static PROPERTIES INSTALL_RPATH "")
+  install(TARGETS ${XBREPLAY_NAME}_static RUNTIME DESTINATION ${XRT_INSTALL_UNWRAPPED_DIR})
+endif()
+
+
+target_link_libraries(${XBREPLAY_NAME}
+  PRIVATE
+  xrt_coreutil
+  ${Boost_FILESYSTEM_LIBRARY}
+  ${Boost_SYSTEM_LIBRARY}
+  ${Boost_PROGRAM_OPTIONS_LIBRARY}
```

Comment:
I recommend not depending on boost::program_options.  Better to do your own parser.  Trying to get rid of boost in general.

Reply:
done